main.cpp

```diff
@@ -14,11 +14,47 @@ Create a branch named Part2
     you should be able to deduce the return type of those functions based on their usage in Person::run()
     You'll need to insert the Person struct from the video in the space below.
  */
+struct Person	
+{	
+	int age;
+	int height;
+	float hairLength;
+	float GPA;
+	unsigned int SATScore;
 
+    struct Foot				
+	{			
+        float distanceFromGround = 0 ;
+        float momentum = 0;
 
+    	int stepForward( float momentum, float distanceFromGround );		
+		float stepSize();		
+	};			
 
+    Foot leftFoot;
+    Foot rightFoot;
 
+    void run( int howFast, bool startWithLeftFoot );
+};
+
+
+int Person::Foot::stepForward( float momentumT0, float distanceFromGroundT0 )				
+{			
+    if ( momentumT0*distanceFromGroundT0 > 0 )
+    
+    return 1;			
+    
+    else return 0 ;
```

Comment:
return statements are the last thing that is executed in a function. 
if the `if()` block is not entered, it is skipped and the next thing that happens is `return 0;`
Since there is no code after the `return 0;`  you do not need the `else` here.
- [ ] revise
- [ ] fix this formatting.
- [ ] add spaces around your operators so they're easier to read `* + =`

Reply:
since stepforward changed back to type void I removed all return statements.   I added the curly braces to add a block for the if sentence to do something.